cmd/snap-seccomp/main.go

```diff
@@ -182,6 +182,23 @@ package main
 // #ifndef PTRACE_SETFPXREGS
 // #define PTRACE_SETFPXREGS 19
 // #endif
+//
+//#include <linux/version.h>
+//#if LINUX_VERSION_CODE < KERNEL_VERSION(3,5,0)
+// /* Define missing kcmp constants */
+// #define KCMP_FILE 0
+// #define KCMP_VM 1
+// #define KCMP_FILES 2
+// #define KCMP_FS 3
+// #define KCMP_SIGHAND 4
+// #define KCMP_IO 5
+// #define KCMP_SYSVSEM 6
+//#else //  LINUX_VERSION_CODE < KERNEL_VERSION(3,5,0)
+// #include <linux/kcmp.h>
+// #if LINUX_VERSION_CODE < KERNEL_VERSION(4,13,0)
+// #define KCMP_EPOLL_TFD 7
+// #endif //  LINUX_VERSION_CODE < KERNEL_VERSION(4,13,0)
+//#endif //  LINUX_VERSION_CODE < KERNEL_VERSION(3,5,0)
```

Comment:
I think you were right to pick 3.19 first time around: while the system call is in 3.5, the `<linux/kcmp.h>` header is not installed (at least it doesn't seem to be in trusty's linux-libc-dev package built from 3.13).

Also as a style note, I'd suggest putting the primary path of these if statements first with fallbacks after. Here we've got the main `#include <linux.kcmp.h> buried in the middle of two lots of fallbacks, so it isn't as obvious what the common case is going to be.